### PR TITLE
[Hotfix] Rewrite force_archive script [OSF-7921]

### PR DESCRIPTION
**Pull request by mfraezz**
## Purpose
Expand/improve functionality of `scripts/force_archive.py`

## Changes
* Rewrite script as management command
* Improve behavior

## Side effects
None expected

## Ticket
[[OSF-7921]](https://openscience.atlassian.net/browse/OSF-7921)

### CODE REVIEW

**Comment by sloria** [comment]
There are a number of flake8 errors in force_archive.py. I've unignored `osf/management` in our flake8 config, so if you update with master, you'll be able to lint the file.

File: osf/management/commands/force_archive.py [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+    if not isinstance(tree, list):
+        tree = [tree]
+    for filenode in tree:
+        if filenode['deleted']:
+            continue
+        file_obj = filenode['object']
+        cloned = file_obj.clone()
+        if cloned.is_deleted:
+            if cloned.is_file:
+                cloned.recast(OsfStorageFile._typedmodels_type)
```
Comment (sloria) [inline osf/management/commands/force_archive.py]: Minor: `._typedmodels_type` is unnecessary, but it's fine.

File: osf/management/commands/force_archive.py [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+            else:
+                cloned.recast(OsfStorageFolder._typedmodels_type)
+        if not parent:
+            parent = reg.get_addon('osfstorage').get_root()
+            cloned.name = 'Archive of OSF Storage'
```
Comment (sloria) [inline osf/management/commands/force_archive.py]: Could you do:

```python
node_settings = reg.get_addon('osfstorage')
parent = node_settings.get_root()
cloned.name = node_settings.archive_folder_name
```

File: osf/management/commands/force_archive.py [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+        cloned.node = reg
+        cloned.copied_from = file_obj
+        try:
+            cloned.save()
+        except IntegrityError:
+            # Files sometimes already created
+            cloned = reg.files.get(name=cloned.name)
+
+        if file_obj.versions.exists() and filenode['version']:  # Min version identifier is 1
+            if not cloned.versions.filter(identifier=filenode['version']).exists():
```
Comment (sloria) [inline osf/management/commands/force_archive.py]: Just curious: In what cases do we hit this branch?

Comment (mfraezz) [inline osf/management/commands/force_archive.py]: Only in the event that a file copy is successful, but the completion of the archive task fails will this be avoided.

File: osf/management/commands/force_archive.py [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+        elif log.action == 'osf_storage_file_removed':
+            # Find parent and add to children, or undelete
+            logger.info('Reverting delete {}:{} from {}'.format(file_obj._id, file_obj.name, log.date))
+            file_tree, _ = modify_file_tree_recursive(file_tree, file_obj, deleted=False, cached=bool(file_obj._id in obj_cache))
+        elif log.action == 'osf_storage_file_updated':
+            # Find file and revert version
+            logger.info('Reverting update {}:{} from {}'.format(file_obj._id, file_obj.name, log.date))
+            file_tree, _ = modify_file_tree_recursive(file_tree, file_obj, deleted=None, cached=bool(file_obj._id in obj_cache))
+            pass
+        elif log.action == 'osf_storage_folder_created':
```
Comment (mfraezz) [inline osf/management/commands/force_archive.py]: Could, but I intentionally kept these separate for logging. I can change that if you don't think it matters

Comment (mfraezz) [inline osf/management/commands/force_archive.py]: As a note to not miss this, this was "outdated" because `pass` was removed.

**Pull request merged**